### ISSUE TITLE
DateTImes in Caliper have more restrictive format

### ISSUE THREAD

**Issue by ViktorHaag** [IMSGlobal/caliper-python#181]
According to the Spec, date-times in Caliper have this requirement:
>  An ISO 8601 date and time value expressed with millisecond precision ... The value MUST be expressed using the format YYYY-MM-DDTHH:mm:ss.SSSZ set to UTC with no offset specified.

This is much more restrictive than the wide array of formats permitted by ISO 8601. The sensor's validation of date-time values input needs to be stricter to match this where required.

**Comment by jonespm** [comment]
Yeah, when we were testing with the Unizin caliper endpoint we had to format the date like

```
event_time = datetime.now()
event_time.strftime('%Y-%m-%dT%H:%M:%S') + event_time.strftime('.%f')[:4] + 'Z'
```

This was from[ this answer ](https://stackoverflow.com/a/52836471/3708872) but nothing a ton prettier that worked. Even the [Zulu Library](https://pypi.org/project/zulu) didn't seem to have any way to output the character 'Z' at the end. It seems like

YYYY-MM-DDTHH:mm:ss.SSS+00:00
and
YYYY-MM-DDTHH:mm:ss.SSSZ

should be the same thing, though I didn't try with +00:00 instead of Z.

**Comment by ViktorHaag** [comment]
I've tried a bunch of different libraries, and the easiest to work with seems to be [pendulum](https://pendulum.eustace.io/docs/). I haven't investigated yet whether it can do strict parsing based on a format string, but for flexible parsing it can read in and attempt to parse a date-time string, and then to get the format required by Caliper:
``` python
import pendulum
dt = pendulum.parse('2016-09-02T11:30:13.111Z')
print(dt.in_tz('UTC').format('YYYY-MM-DDTHH:MM:ss.SSS')+'Z')
>> '2016-09-02T11:09:13.111Z'
```

If we had a bit more flexibility, pendulum can do what's needed without the format string and get 'the right time', but to get _exactly three digits of millisecond precision plus the Z notifier_, you have to use string concat tricks like this.

**Comment by ViktorHaag** [comment]
Addressed with PR #182. I think I'd rather not float that fix into the master branch since it's a functional change (albeit a defect fix). We'll keep it in develop so that it gets released with the Caliper v1.2 release.

**Comment by ViktorHaag** [comment]
Addressed with #182
